linux/deploy_vm/rebuild_unattend_install_iso.yml

```diff
@@ -21,245 +21,254 @@
-      ansible.builtin.set_fact:
-        iso_files_to_update:
-          - '/boot/grub/grub.cfg'
-          - '/md5sum.txt'
-
-    - name: "Check whether BIOS boot config file isolinux/txt.cfg exists"
-      ansible.builtin.stat:
-        path: "{{ tmp_iso_mount_dir }}/isolinux/txt.cfg"
-      register: isolinux_cfg_stat
-
-    - name: "Add BIOS boot config file for updating"
-      ansible.builtin.set_fact:
-        iso_files_to_update: "{{ iso_files_to_update + ['/isolinux/txt.cfg'] }}"
-      when:
-        - isolinux_cfg_stat is defined
-        - isolinux_cfg_stat.stat is defined
-        - isolinux_cfg_stat.stat.exists is defined
-        - isolinux_cfg_stat.stat.exists
-
-    - name: "Add command options to rebuild unattended install ISO for Ubuntu live server"
-      ansible.builtin.set_fact:
-        rebuild_iso_cmd: "{{ rebuild_iso_cmd }} -map {{ unattend_iso_cache }}/{{ item | basename }} '{{ item }}'"
-      with_items: "{{ iso_files_to_update }}"
-  when: unattend_install_conf is match('Ubuntu/Server')
-
-- name: "Add command options for boot images to rebuild unattend install ISO"
-  ansible.builtin.set_fact:
-    rebuild_iso_cmd: "{{ rebuild_iso_cmd }} {{ bios_boot_image }} -boot_image any next {{ efi_boot_image }}"
-
-- name: "Copy unattend install config files to {{ unattend_iso_cache }}"
-  ansible.builtin.copy:
-    src: "{{ tmp_iso_mount_dir }}{{ item }}"
-    dest: "{{ unattend_iso_cache }}"
-    force: true
-    mode: 0644
-  with_items: "{{ iso_files_to_update }}"
+    - name: Extract specific files inside ISO
+      community.general.iso_extract:
+        image: "{{ src_iso_file_path }}"
+        dest: "{{ src_iso_file_dir }}"
+        files:
+          - 'boot/grub/grub.cfg'
+          - 'isolinux/isolinux.cfg'
+          - 'isolinux/gtk.cfg'
+          - 'md5sum.txt'
 
-- block:
     - name: "Update timeout for boot menu"
       ansible.builtin.replace:
-        path: "{{ unattend_iso_cache }}/isolinux.cfg"
+        path: "{{ src_iso_file_dir }}/isolinux.cfg"
         regexp: "timeout 0"
         replace: "timeout 1"
-      when: >
-        unattend_install_conf is match('Photon') or
-        unattend_install_conf is match('Debian')
 
-    # Update autoinstall config files for Photon
-    - block:
-        - name: "Update boot menu with kickstart for Photon OS"
-          ansible.builtin.replace:
-            path: "{{ unattend_iso_cache }}/{{ item }}"
-            regexp: '(.*)(root=[^ ]+)(.*)'
-            replace: "\\1\\2 ks=cdrom:/isolinux/{{ unattend_install_file_name }} \\3"
-          with_items:
-            - "menu.cfg"
-            - "grub.cfg"
-      when: unattend_install_conf is match('Photon')
+    - name: "Set fact of Debian install type"
+      ansible.builtin.set_fact:
+        debian_install_type: |-
+          {%- if src_iso_file_name.find('amd64') != -1 -%}install.amd
+          {%- else -%}install.386
+          {%- endif -%}
```

Comment:
I think here can get debian install type by searching 'gtk/initrd.gz' in md5sum.txt, it will be more reliable than checking iso file name.

For example,

32bit: 
```
525813f2e18fd7845d5ccf2dfbe96ef7  ./install.386/gtk/initrd.gz
```

64bit:
```
80c12152218bd83cf3cd449b33beb6bc  ./install.amd/gtk/initrd.gz
```
```

Reply:
Thanks. Done